File: python-edition/solids/ceiling.py
from raylib import *
import pyray as p
from gameio.dataio import DataIO

class Ceiling:
    def __init__(self, x,y,w,h) -> None:
        self.palette = DataIO().retrievePalette()
        self.ceiling = p.Rectangle(x,y,w,h)
    def Draw(self):
        DrawRectangleRec(self.ceiling, self.palette['brickred'])
    def IsPlayerUnder(self,player):
        if player.rec.x < self.ceiling.x + self.ceiling.width and player.rec.x + player.rec.width > self.ceiling.x and player.rec.y > self.ceiling.y:
            return True #player is under ceiling
        else:
            return False

File: swift-edition/liner-swift-edition/liner-swift-edition/solids/ceiling.swift
//
//  ceiling.swift
//  liner-swift-edition
//
//  Created by easontek2398 on 8/6/22.
//

import Foundation
import Raylib

class Ceiling {
    var ceiling: Rectangle = Rectangle()
    init(x posx: Float, y posy: Float, width Width: Float, height Height: Float) {
        self.ceiling = Rectangle(x: posx, y: posy, width: Width, height: Height)
    }
    
    func Draw() {
        
    }
    func IsPlayerUnder(_ player: Player) -> Bool {
        if player.rec!.x < self.ceiling.x + self.ceiling.width && player.rec!.x + player.rec!.width > self.ceiling.x && player.rec!.y > self.ceiling.y {
            return true
        } else {
            return false
        }
    }
}
